docroot/modules/custom/sitenow_signage/sitenow_signage.module

```diff
@@ -446,3 +681,68 @@ function sitenow_signage_outlook_resource_allowed_values() {
   $api_client = \Drupal::service('uiowa_hours.api');
   return $api_client->getResources($group) ?: ['none' => 'No resources available'];
 }
+
+/**
+ * Helper function to get sign orientation.
+ *
+ * @param Drupal\paragraphs\Entity\Paragraph $paragraph
+ *   A $paragraph object.
+ *
+ * @return string
+ *   Return the orientation string.
+ *
+ * @todo https://github.com/uiowa/uiowa/issues/5028
+ */
+function sitenow_signage_get_orientation(Paragraph $paragraph) {
+
+  // Need to traverse up to the sign node to get orientation if available.
+  $parent_slide = $paragraph->getParentEntity();
+  $referring_item = $parent_slide->_referringItem;
+  $orientation = 'horizontal';
+  if ($referring_item) {
+    /** @var \Drupal\Core\Field\EntityReferenceFieldItemList $referring_field */
+    $referring_field = $referring_item->getParent();
+    if ($referring_field) {
+      $parent_entity = $referring_field->getParent();
+      // Add cache tag for the sign node so table
+      // updates when sign orientation changes.
+      $variables['#cache']['tags'][] = 'node:' . $parent_entity->get('nid')->value;
+      if ($parent_entity->get('field_sign_orientation')->value == 'vertical') {
+        $orientation = 'vertical';
+      }
+    }
+  }
+
+  return $orientation;
+}
+
+/**
+ * Fast object to array conversion function.
+ *
+ * Helper function to convert objects recursively to arrays.
+ * This is roughly 1.5-2x faster than json_decode() + json_encode().
+ *
+ * @param mixed $obj
+ *   Any object.
+ *
+ * @return array|mixed
+ *   Returns an object that is being converted into an array
+ *   until the complete object is only an array, then it
+ *   returns an array.
+ */
+function sitenow_signage_object_to_array(mixed $obj): mixed {
```

Comment:
I'm a bit skeptical that this is necessary. Seems like something that might already be handled by a core Drupal utility or helper function.

Reply:
I'm not opposed to introducing a new helper function to handle this if there is not something that already exists that does the same thing. But let's not bury it in the signage module where it is unable to be used by the rest of the platform.

Reply:
I propose moving it to `uiowa_core` then, if that's okay with you?

Reply:
I'm good with this for now. Thinking long term it might be better as a trait.